src/OrleansProviders/Streams/Common/PooledCache/ObjectPool.cs

```diff
@@ -13,14 +13,14 @@ public class ObjectPool<T> : IObjectPool<T>
     {
         private const int DefaultPoolCapacity = 1 << 10; // 1k
         private readonly Stack<T> pool;
-        private readonly Func<IObjectPool<T>, T> factoryFunc;
+        private readonly Func<T> factoryFunc;
 
         /// <summary>
         /// Simple object pool
         /// </summary>
         /// <param name="factoryFunc"></param>
         /// <param name="initialCapacity"></param>
-        public ObjectPool(Func<IObjectPool<T>, T> factoryFunc, int initialCapacity = DefaultPoolCapacity)
+        public ObjectPool(Func<T> factoryFunc, int initialCapacity = DefaultPoolCapacity)
```

Comment:
I think the <code>factoryFunc</code> is pretty self-documenting, and the root cause of such change is located here - https://github.com/dotnet/orleans/blob/master/src/OrleansProviders/Streams/Common/PooledCache/IObjectPool.cs#L64